spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -104,30 +144,100 @@
         }
       }
 
-      if (splitSize != null) {
-        scan = scan.option(TableProperties.SPLIT_SIZE, splitSize.toString());
+      for (Expression filter : filterExpressions()) {
+        scan = scan.filter(filter);
       }
 
-      if (splitLookback != null) {
-        scan = scan.option(TableProperties.SPLIT_LOOKBACK, splitLookback.toString());
+      try (CloseableIterable<FileScanTask> filesIterable = scan.planFiles()) {
+        this.files = Lists.newArrayList(filesIterable);
+      } catch (IOException e) {
+        throw new UncheckedIOException("Failed to close table scan: " + scan, e);
       }
+    }
 
-      if (splitOpenFileCost != null) {
-        scan = scan.option(TableProperties.SPLIT_OPEN_FILE_COST, splitOpenFileCost.toString());
-      }
+    return files;
+  }
 
-      for (Expression filter : filterExpressions()) {
-        scan = scan.filter(filter);
+  @Override
+  protected List<CombinedScanTask> tasks() {
+    if (tasks == null) {
+      CloseableIterable<FileScanTask> splitFiles = TableScanUtil.splitFiles(
+          CloseableIterable.withNoopClose(files()),
+          splitSize);
+      CloseableIterable<CombinedScanTask> scanTasks = TableScanUtil.planTasks(
+          splitFiles, splitSize,
+          splitLookback, splitOpenFileCost);
+      tasks = Lists.newArrayList(scanTasks);
+    }
+
+    return tasks;
+  }
+
+  @Override
+  public NamedReference[] filterAttributes() {
+    Set<Integer> partitionFieldSourceIds = Sets.newHashSet();
+
+    for (Integer specId : specIds()) {
+      PartitionSpec spec = table().specs().get(specId);
+      for (PartitionField field : spec.fields()) {
+        partitionFieldSourceIds.add(field.sourceId());
       }
+    }
 
-      try (CloseableIterable<CombinedScanTask> tasksIterable = scan.planTasks()) {
-        this.tasks = Lists.newArrayList(tasksIterable);
-      }  catch (IOException e) {
-        throw new RuntimeIOException(e, "Failed to close table scan: %s", scan);
+    Map<Integer, String> nameById = TypeUtil.indexQuotedNameById(table().schema().asStruct());
```

Comment:
That possible but the resolution is done using column IDs so we should be fine.

Reply:
So I was more asking about table().schema().asStruct(), that's only the current schema, and will have the wrong map for this case?   As we are looking through all historical partition-spec, some of which may have been in time when partition field had a different name.

Reply:
Do you have a specific example in mind, @szehon-ho? I can add it to the test cases.

Reply:
Initially I was fearing a perverse case where you rename partition columns to be each other (then. you would filter on the wrong column), but Iceberg seems to prevent it:
* Caused by: java.lang.IllegalArgumentException: Cannot create identity partition sourced from different field in schema: b
* java.lang.IllegalArgumentException: Cannot use partition name more than once: b

Then, something simple like:

* Have a table partitioned by a, insert data.
* Rename the partition column to b (table.schema() will now be different schema)
* Run the query

In this case, the namedAttribute returned will be different than what's on the data.  I guess here it's just a missed opportunity than a correctness issue.  Could be taken advantage of by checking all values for table.schemas(), but probably it's worth a separate pr as may be something to do carefully.

But I may be missing something, let me know if I am.

Reply:
I don't think there's a problem here. Iceberg is going to use a consistent set of names for a table that is determined at table load time. And Iceberg metadata is tracked using field IDs, so even if a name changes it's okay.

If you have a table partitioned by 1:a and change that to b in the schema, then you have a table partitioned by 1:b. Because only one schema is reported to Spark, you'll either use a or b and when Spark passes the filter back you'll consistently get a or b in the predicate.

Reply:
Yeah, @rdblue is right. I'll add a test for the rename to be safe.

Reply:
Added a test when the source column is renamed.